native-static-images/README.md

```diff
@@ -0,0 +1,111 @@
+# Building a Native Executable as Statically Linked or Mostly-Statically Linked.
+
+GraalVM Native Image by default builds dynamically linked binaries: at build time it loads your application classes and interfaces and hooks them together in a process of dynamic linking.
+
+However, you can create a statically linked or mostly-static linked native executable, depending on your needs. 
+
+**A static native executable** is a statically linked binary that can be used without any additional library dependencies.
+A static native executable is easy to distribute and deploy on a slim or distroless container (a scratch container).
+You can create a static native executable by statically linking it against [musl-libc](https://musl.libc.org/), a lightweight, fast and simple `libc` implementation.
+
+**A mostly-static native executable** is a binary that links everything (`zlib`, JDK shared libraries) except the standard C library, `libc`. This is an alternative option to staticly linking everything. Also, depending on the user's code, it may link `libstdc+` and `libgcc`.
+This approach is ideal for deployment on a distroless container image.
+
+> Note: This currently only works when linked against `libc`.
+
+This guide shows how you can take advantage of Native Image linking options including fully dynamic, fully static, and mostly static (except `libc`) to generate an executable ideal for your deployment scenario.
+
+For the following guide you should be have:
+
+- Linux AMD64 operating system
+- GraalVM distribution for Java 11 or higher.
+- A 64-bit `musl` toolchain, `make`, and `configure`
+- The latest `zlib` library
+
+## Preparation
+
+1. Download and install the latest GraalVM JDK with Native Image using the [GraalVM JDK Downloader](https://github.com/graalvm/graalvm-jdk-downloader):
+    ```bash
+    bash <(curl -sL https://get.graalvm.org/jdk) -c 'native-image' 
+    ```
+
+As a preparation step, you need install the `musl` toolchain, compile and install `zlib` into the toolchain.
```

Comment:
To meet the prerequisites, you need to install the `musl` toolchain, compile and install `zlib` into the toolchain.